src/lib.rs

```diff
@@ -340,8 +341,8 @@ mod test {
                 // We found an unused port.
                 Ok(socket) => return socket.local_addr().unwrap(),
                 Err(e) => match e.kind() {
-                    std::io::ErrorKind::AddrInUse => {
-                        // We hit a used port, try again.
+                    ErrorKind::AddrInUse | ErrorKind::PermissionDenied => {
```

Comment:
Did you pick up an extra change somewhere?

Reply:
Yes, this was something I hit while testing. See first post. Should I factor this out?

Reply:
I don't mind; hence the approval.  But it's going to be hard to identify, so maybe that is the right thing to do.

Reply:
I split this out into #24